src/ray/raylet/node_manager.cc

```diff
@@ -1845,6 +1845,36 @@ void NodeManager::HandleRequestWorkerLease(const rpc::RequestWorkerLeaseRequest
   SubmitTask(task, Lineage());
 }
 
+void NodeManager::HandleRequestResourceLease(
+    const rpc::RequestResourceLeaseRequest &request,
+    rpc::RequestResourceLeaseReply *reply, rpc::SendReplyCallback send_reply_callback) {
+  // rpc::Bundle bundle_spec;
+  auto bundle_spec = BundleSpecification(request.bundle_spec());
+  RAY_LOG(DEBUG) << "bundle lease request " << bundle_spec.BundleId();
+  auto resource_ids = ScheduleBundle(cluster_resource_map_, bundle_spec);
+  if (resource_ids.AvailableResources().size() == 0) {
+    send_reply_callback(Status::OutOfMemory("reserve resource failed"), nullptr, nullptr);
```

Comment:
This should be returned as a boolean field in the proto message indicating failure, not a system-level OOM exception.

Reply:
Change in the https://github.com/ray-project/ray/pull/9039